capa/features/extractors/viv/insn.py

```diff
@@ -120,6 +122,35 @@ def extract_insn_api_features(fh: FunctionHandle, bb, ih: InsnHandle) -> Iterato
                 yield API(name[1:]), ih.address
             return
 
+        if f.vw.metadata["Format"] == "elf":
+            if "SymbolTable" not in f.vw.metadata:
+                # the symbol table gets stored as a function's attribute in order to avoid running
+                # this code everytime the call is made, thus preventing the computational overhead.
+                elf = f.vw.parsedbin
+                endian = "<" if elf.getEndian() == 0 else ">"
+                bitness = elf.bits
+
+                SHT_SYMTAB = 0x2
+                for section in elf.sections:
+                    if section.vsGetField("sh_info") & SHT_SYMTAB != 0:
+                        strtab = elf.sections[section.vsGetField("sh_link")]
+                        sh_symtab = Shdr.from_viv(section, elf.getSectionBytes(section.name))
+                        sh_strtab = Shdr.from_viv(strtab, elf.getSectionBytes(strtab.name))
+
+                symtab = SymTab(endian, bitness, sh_symtab, sh_strtab)
+                f.vw.metadata["SymbolTable"] = symtab
+
+            symtab = f.vw.metadata["SymbolTable"]
+            for symbol in symtab.get_symbols():
+                sym_name = symtab.get_name(symbol)
+                sym_value = symbol.value
+                sym_info = symbol.info
+
+                STT_FUNC = 0x2
+                if sym_value == target and sym_info & STT_FUNC != 0:
+                    yield API(sym_name), ih.address
+                    yield FunctionName(sym_name), ih.address
```

Comment:
the `FunctionName` feature should be extracted for the target address, not the caller address. `FunctionName` is used to describe a known name for a function, so that we can reference it in rules, like "look for this thing, as long as its not found in a memcpy implementation". 

probably this should be extracted in another function-scoped analysis pass in this file: https://github.com/mandiant/capa/blob/master/capa/features/extractors/viv/function.py